Demo/Demo/ViewModels/BaseViewModel.swift

```diff
@@ -143,22 +155,13 @@ class BaseViewModel: ObservableObject, PayPalWebCheckoutDelegate, CardDelegate,
         cardClient.approveOrder(request: cardRequest, context: context)
     }
 
-    func checkoutWithNativeClient(orderId: String?) async throws -> NativeCheckoutResult {
+    func checkoutWithNativeClient(orderId: String?) async throws {
         guard let orderId = orderId else {
-
-            return NativeCheckoutResult.error(CoreSDKError(code: 0, domain: "Order Id is null", errorDescription: "Order Id is null"))
+            return
         }
         let nativeCheckoutClient = try await getNativeCheckoutClient()
-        let paypalRequest = PayPalRequest(orderID: orderId)
         nativeCheckoutClient.delegate = self
-        DispatchQueue.main.async {
-            nativeCheckoutClient.start(request: paypalRequest)
-        }
-        guard let nativeCheckoutResult = nativeCheckoutResult
-        else {
-            throw CoreSDKError(code: 0, domain: "checkout", errorDescription: "didn't get result from checkout")
-        }
-        return nativeCheckoutResult
+        nativeCheckoutClient.start(presentingViewController: nil, orderID: orderId, deleagate: self)
```

Comment:
We can default `presentingViewController` to `nil` in the start function so merchants can omit `presentingViewController` entirely if they don't need it. The this can be updated to:
```
nativeCheckoutClient.start(orderID: orderID, deleagate: self)
```